### DISCUSSION TITLE
Populating form with previous submission

### DISCUSSION THREAD

**Discussion by kjkent** [rjsf-team/react-jsonschema-form#2825] [Q&A] [answered]
Hi there, and thanks for developing such a useful tool.

I'm building a set of forms that a user will likely return to after submission. When a user returns to a form, is it possible to have that form prepopulated with their previous submission? I thought of fetching the info from the DB with a GET request, but I'm unsure of how to then insert this data into the form. If you have a suggestion for how I might accomplish this, I'd greatly appreciate your input (pun intended ;).

Many thanks.

**Comment by nickgros** [comment, marked as answer]
@kjkent You could treat `Form` as a controlled component, and pass the previous submission as the `formData` prop. You would have to ensure you update `formData` using the `onChange` prop. See [Quick Start -> Controlled Component](https://react-jsonschema-form.readthedocs.io/en/latest/quickstart/#controlled-component).

**Reply by kjkent** [reply]
Thank you; this is perfect.
